api/ask_astro/chains/custom_llm_output_lines_parser.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional
+
+from langchain.output_parsers.pydantic import PydanticOutputParser
+from langchain.retrievers.multi_query import LineList
+
+
+class CustomLineListOutputParser(PydanticOutputParser):
+    """
+    Output parser for a list of lines with additional cleaning and fail checks.
```

Comment:
see original implementation of LineListOutputParser from LangChain here: https://api.python.langchain.com/en/latest/_modules/langchain/retrievers/multi_query.html#